src/EditorFeatures/CSharpTest/BlockCommentEditing/BlockCommentEditingTests.cs

```diff
@@ -0,0 +1,563 @@
+        public async Task BoundCheckInsertOnEndLine()
+        {
+            var code = @"
+    /*
+     *$$/";
+            var expected = @"
+    /*
+     *
+     *$$/";
+            await VerifyAsync(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.BlockCommentEditing)]
+        public async Task InsertOnStartLine2_Tab()
+        {
+            var code = @"
+    /*$$<tab>*/
+";
+            var expected = @"
+    /*
+     *$$<tab>*/
+";
+            await VerifyTabsAsync(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.BlockCommentEditing)]
+        public async Task InsertOnStartLine3_Tab()
+        {
+            var code = @"
+    /*<tab>$$<tab>1.
+     */
+";
+            var expected = @"
+    /*<tab>
+     *<tab>$$<tab>1.
+     */
+";
+            await VerifyTabsAsync(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.BlockCommentEditing)]
+        public async Task InsertOnStartLine4_Tab()
+        {
+            var code = @"
+    /* <tab>1.$$
+     */
+";
+            var expected = @"
+    /* <tab>1.
+     * <tab>$$
+     */
+";
+            await VerifyTabsAsync(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.BlockCommentEditing)]
+        public async Task InsertOnStartLine6_Tab()
+        {
+            var code = @"
+    /*<tab>$$
+";
+            var expected = @"
+    /*<tab>
+     *<tab>$$
+";
+            await VerifyTabsAsync(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.BlockCommentEditing)]
+        public async Task InsertOnMiddleLine2_Tab()
+        {
+            var code = @"
+    /*
+     *$$<tab>*/
+";
+            var expected = @"
+    /*
+     *
+     *$$<tab>*/
+";
+            await VerifyTabsAsync(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.BlockCommentEditing)]
+        public async Task InsertOnMiddleLine3_Tab()
+        {
+            var code = @"
+    /*
+     * $$<tab>1.
+     */
+";
+            var expected = @"
+    /*
+     * 
+     * $$<tab>1.
+     */
+";
+            await VerifyTabsAsync(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.BlockCommentEditing)]
+        public async Task InsertOnMiddleLine4_Tab()
+        {
+            var code = @"
+    /*
+     * <tab>1.$$
+     */
+";
+            var expected = @"
+    /*
+     * <tab>1.
+     * <tab>$$
+     */
+";
+            await VerifyTabsAsync(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.BlockCommentEditing)]
+        public async Task InsertOnMiddleLine5_Tab()
+        {
+            var code = @"
+    /*
+     *<tab> 1.
+     *<tab> $$
+     */
+";
+            var expected = @"
+    /*
+     *<tab> 1.
+     *<tab> 
+     *<tab> $$
+     */
+";
+            await VerifyTabsAsync(code, expected);
```

Comment:
The spaces/tabs before `*` are not handled by this command handler. It simply calls `editorOperations.InsertNewLine();` and expected the caret magically aligns.

OK.